activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -134,7 +134,10 @@ def first!
     #
     #   Person.last # returns the last object fetched by SELECT * FROM people
     #   Person.where(["user_name = ?", user_name]).last
-    #   Person.order("created_on DESC").offset(5).last
+    #   Person.order("created_on DESC").offset(5).last # returns the last object (or nil if Person.all.size <= 5)
+    #
+    #  Note that the use of offset with last is deprecated, because relations with an offset cannot be automatically reversed.
```

Comment:
The deprecation warning in that `rescue` block is incorrect here. The warning should explicitly say what is wrong (offset being applied)